obal/data/tasks/download_and_extract_source.yml

```diff
@@ -8,68 +8,6 @@
     state: 'directory'
     path: "{{ obal_tmp_dir }}"
 
-- block:
-    - name: 'Extract upstream directory'
-      set_fact:
-        upstream_directory: "{{ upstream.split('/')[-1] }}"
-
-    - name: 'Clone upstream'
-      git:
-        repo: "{{ upstream }}"
-        dest: "{{ obal_tmp_dir }}/{{ upstream_directory }}"
-        version: "{{ branch }}"
-        update: yes
-
-    - name: 'Copy upstream files'
-      synchronize:
-        src: "{{ obal_tmp_dir }}/{{ upstream_directory }}/{{ item }}"
-        dest: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}/"
-      with_items: "{{ upstream_files }}"
-  when: #block
-    - upstream.endswith('.git')
-    - downstream_version is not defined
-  rescue: #block
-    - name: 'Remove package directory'
-      file:
-        state: 'absent'
-        path: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}"
-
-- block:
-    - name: 'Make package download directory'
-      file:
-        state: 'directory'
-        path: "{{ obal_tmp_dir }}/{{ inventory_hostname }}"
-
-    - name: 'Fetch srpm'
-      get_url:
-        url: "{{ upstream }}"
-        dest: "{{ obal_tmp_dir }}/{{ inventory_hostname }}/"
-
-    - name: 'Find SRPM'
-      find:
-        pattern: "*.src.rpm"
-        paths: "{{ obal_tmp_dir }}/{{ inventory_hostname }}/"
-      register: "srpm"
-
-    - name: 'Explode srpm'
-      shell: "rpm2cpio {{ srpm.files[0].path | basename }} | cpio -idmcv --no-absolute-filenames"
-      args:
-        chdir: "{{ obal_tmp_dir }}/{{ inventory_hostname }}"
-
-    - name: 'Copy upstream files'
-      synchronize:
-        src: "{{ obal_tmp_dir }}/{{ inventory_hostname }}/{{ item }}"
-        dest: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}/"
-      with_items: "{{ upstream_files }}"
-  when: #block
-    - upstream.endswith('src.rpm')
-    - downstream_version is not defined
-  rescue: #block
-    - name: 'Remove package directory'
-      file:
-        state: 'absent'
-        path: "./{{ package_base_dir }}/{{ inventory_hostname }}"
-
 - block:
```

Comment:
I don't have experience with ```rpmdev-bumpspec``` but that's supposed to do pretty much the same: ```rpmdev-bumpspec --comment 'Release {{ inventory_hostname }} {{ version }}' --new '{{ version }}' {{ spec_file }}```

On the other hand, here you also try to replace globals which rpmdev-bumpspec can only do with release but not versions.

Reply:
Yea, we have some specs that go outside the norm using globals :/